File: Laba5/Laba2.5/Laba2.5/parallelogram.cpp
#include "parallelogram.h"

double parallelogram::countArea() { // Обрахунок площі
	double Side1, Side2, sina;
	Side1 = sqrt(pow(V[1].x - V[0].x, 2) + pow(V[1].y - V[0].y, 2)); // Перша сторона
	Side2 = sqrt(pow(V[2].x - V[1].x, 2) + pow(V[2].y - V[1].y, 2)); // Друга сторона
	sina = sin(acos(abs(((V[0].x - V[1].x)*(V[2].x - V[1].x) + (V[0].y - V[1].y)*(V[2].y - V[1].y))/(Side1*Side2)))); // Синус кута між сторонами
	return Side1 * Side2 * sina;
}

File: Laba5/OP2,5Python/OP2,5Python/parallelogram.py
from TQuadrangle import TQuadrangle
import math
class Parallelogram(TQuadrangle):
    def __init__(self, Coords):
        super().__init__(Coords, "Parallelogram")
        self.area = self.countArea()

    def countArea(self):
        Side1 = math.sqrt(pow(self.Coords[1].x - self.Coords[0].x, 2) + pow(self.Coords[1].y - self.Coords[0].y, 2))
        Side2 = math.sqrt(pow(self.Coords[2].x - self.Coords[1].x, 2) + pow(self.Coords[2].y - self.Coords[1].y, 2))
        sina = math.sin(math.acos(math.fabs(((self.Coords[0].x - self.Coords[1].x)*(self.Coords[2].x - self.Coords[1].x) + (self.Coords[0].y - self.Coords[1].y)*(self.Coords[2].y - self.Coords[1].y))/(Side1*Side2))))
        return sina * Side1 * Side2